pallets/ds-maps/src/lib.rs

```diff
@@ -0,0 +1,183 @@
+    type WeightInfo: WeightInfo;
+    // new types, fixes required
+    type Coordinates: Parameter;
+    ///guess use u32 for representing global coords, u16 for local
+    type CoordinateSize: Default + Parameter;
+}    
+pub trait WeightInfo {
+    fn register_zone() -> Weight;
+}
+
+decl_storage!{
+    // A unique name is used to ensure that the pallet's storage items are isolated.
+    // This name may be updated, but each pallet in the runtime must use a unique name.
+    // ---------------------------------vvvvvvvvvvvvvv
+    trait Store for Module<T: Trait> as DSMapsModule {
+        //MAX is 4_294_967_295. Change if required more.
+        TotalBoxes get(fn total_boxes): u32;    
+        
+        CityMap get(fn map_data): 
+            map hasher(blake2_128_concat) u32 => ZoneOf<T>;
+    }
+}
+pub type ZoneOf<T> = Zone<<T as Trait>::CoordinateSize>;
+
+// Pallets use events to inform users when important changes are made.
+// https://substrate.dev/docs/en/knowledgebase/runtime/events
+decl_event!(
+    pub enum Event<T>
+    where
+        AccountId = <T as frame_system::Trait>::AccountId,
+        CoordinateSize = <T as Trait>::CoordinateSize,
+    {
+        // Event documentation should end with an array that provides descriptive names for event parameters.
+        ///TODO add more meta/remove
+        MapInitialized(CoordinateSize),
+        /// New account has been created [zone number, its type], TODO later add printing coords
+        ZoneCreated(u32, AccountId, ZoneType),
+    }
+);
+
+// Errors inform users that something went wrong.
+// learn more https://substrate.dev/docs/en/knowledgebase/runtime/errors
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        /// Error names should be descriptive.
+        NoneValue,
+        /// Operation is not valid
+        InvalidAction,
+        /// Incorrect data provided
+        InvalidData,
+        /// Origin do not have sufficient privileges to perform the operation
+        NotAuthorized,
+        /// Account doesn't exist
+        NotExists,
+        // add additional errors below
+    }
+}
+
+// Dispatchable functions allows users to interact with the pallet and invoke state changes.
+// These functions materialize as "extrinsics", which are often compared to transactions.
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        // Errors must be initialized if they are used by the pallet.
+        type Error = Error<T>;
+
+        // Events must be initialized if they are used by the pallet.
+        fn deposit_event() = default;
+
+        #[weight = <T as Trait>::WeightInfo::register_zone()]
+        pub fn zone_add(origin, 
+                        zone_type: ZoneType, 
+                        points: [T::CoordinateSize; 6]) -> dispatch::DispatchResult {
```

Comment:
Again, it is not clear what are these "points", it is better to have `bounding_box: Box3D<T::Coord3D>` here.